Wikipedia/UI-V5/UIViewController+WMFSearchButton.m

```diff
@@ -0,0 +1,51 @@
+//
+//  UIViewController+WMFSearchButton.m
+//  Wikipedia
+//
+//  Created by Brian Gerstle on 9/30/15.
+//  Copyright © 2015 Wikimedia Foundation. All rights reserved.
+//
+
+#import "UIViewController+WMFSearchButton.h"
+#import "WMFSearchViewController.h"
+#import "UIBarButtonItem+WMFButtonConvenience.h"
+#import "WMFArticleContainerViewController.h"
+#import "SessionSingleton.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation UIViewController (WMFSearchButton)
+
+- (UIBarButtonItem*)wmf_searchBarButtonItemWithDelegate:(UIViewController<WMFSearchPresentationDelegate>*)delegate {
+    @weakify(self);
+    @weakify(delegate);
+    return [UIBarButtonItem wmf_buttonType:WMFButtonTypeMagnify handler:^(id sender) {
+        @strongify(self);
+        @strongify(delegate);
+        if (!delegate || !self) {
+            return;
+        }
+
+        MWKSite* searchSite;
+        if ([delegate respondsToSelector:@selector(searchSite)]) {
+            searchSite = [delegate searchSite];
+        } else {
+            // if the delegate doesn't have a specific site we should search from, default to the user's setting
+            searchSite = [[SessionSingleton sharedInstance] searchSite];
+        }
+
+        WMFSearchViewController* searchVC =
+            [WMFSearchViewController searchViewControllerWithSite:searchSite
+                                                        dataStore:[delegate dataStore]];
+        searchVC.searchResultDelegate = delegate;
+        [self wmf_presentSearchViewController:searchVC];
+    }];
+}
+
+- (void)wmf_presentSearchViewController:(UIViewController*)searchViewController {
```

Comment:
Just replacing one line - just present the view controller - 3 less lines of code and no confusing "special" way to present search - its presented the same way as any other VC